File: src/SSHServer/Packets/Unimplemented.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer.Packets
{
    public class Unimplemented : Packet
    {
        public override PacketType PacketType
        {
            get
            {
                return PacketType.SSH_MSG_UNIMPLEMENTED;
            }
        }

        public uint RejectedPacketNumber { get; set; }

        public override void Load(ByteReader reader)
        {
            RejectedPacketNumber = reader.GetUInt32();
        }

        protected override void InternalGetBytes(ByteWriter writer)
        {
            // uint32 packet sequence number of rejected message
            writer.WriteUInt32(RejectedPacketNumber);
        }
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/Packets/Unimplemented.ts
import { Packet } from "./Packet";
import { PacketType } from "./PacketType";
import { ByteReader } from "../ByteReader";
import { ByteWriter } from "../ByteWriter";

export class Unimplemented extends Packet {
    public rejectedPacketNumber: number;

    public getPacketType(): PacketType {
        return PacketType.SSH_MSG_UNIMPLEMENTED;
    }

    protected internalGetBytes(writer: ByteWriter) {
        writer.writeUInt32(this.rejectedPacketNumber);
    }

    public load(reader: ByteReader) {
        this.rejectedPacketNumber = reader.getUInt32();
    }
}
